kv/etcd/store.go

```diff
@@ -355,22 +355,63 @@ func (c *client) Watch(key string) (kv.ValueWatch, error) {
 	return w, err
 }
 
-func (c *client) update(key string) error {
+func (c *client) getFromKVStore(key string) (kv.Value, error) {
 	var (
-		newValue kv.Value
-		err      error
+		nv  kv.Value
+		err error
 	)
-	// we need retry here because if Get() failed on an watch update,
-	// it has to wait 10 mins to be notified to try again
 	if execErr := c.retrier.Attempt(func() error {
-		newValue, err = c.get(key)
+		nv, err = c.get(key)
 		if err == kv.ErrNotFound {
 			// do not retry on ErrNotFound
 			return xretry.NonRetryableError(err)
 		}
 		return err
 	}); execErr != nil && xerrors.GetInnerNonRetryableError(execErr) != kv.ErrNotFound {
-		return execErr
+		return nil, execErr
+	}
+
+	return nv, err
```

Comment:
hmm, interesting, should return nv, nil here? or ErrNotFound will be returned and cause the watches not getting deletion updates? This case should be covered in a test